dashboard/app/models/user_level.rb

```diff
@@ -24,6 +26,10 @@ class UserLevel < ActiveRecord::Base
   belongs_to :level
   belongs_to :script
 
+  scope :attempted, -> { where.not(best_result: nil) }
+  scope :passed, -> { where('best_result >= ?', ActivityConstants::MINIMUM_PASS_RESULT) }
```

Comment:
There are naming inconsistencies, I did notice them. Notice the tense changes between the existing "submitted", "finished", "passing", and "best". Though "passing" makes sense, I don't think "submitting" or "finishing" does, so I elected for the past tense. Ideally, all of them would be changed to past tense verbs, though "best" doesn't quite fit that mold.